### PR TITLE
Add support for Voting Progress

### PR DESCRIPTION
**Pull request by willm30**
This PR adds support for the VotingProgress component: https://github.com/JoinColony/colonyCDapp/pull/401

Specifically: 

1. Stores voting rep params in db on initialisiation
2. Adds skillRep and repSubmitted to motion data

### CODE REVIEW

File: src/handlers/extensions/helpers.ts [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+};
+
+export const addVotingReputationParamsToDB = async (
+  extensionAddress: string,
+  colonyAddress: string,
+): Promise<void> => {
+  await mutate('updateColonyExtensionByAddress', {
+    input: {
+      id: extensionAddress,
+      params: await getVotingReputationParams(colonyAddress),
```
Comment (rdig) [inline src/handlers/extensions/helpers.ts:47]: I'm not a fan of this pattern personally. I'd have this assigned to a variable then pass that along to the mutation input.

**Review by rdig — APPROVED** [review]
Fine by me. Left a comment, but it's more of a nitpick rather than something that needs to be addressed

File: src/trackExtensions.ts [hunk 2]
```diff
@@ -1,5 +1,7 @@
 import { Extension, getExtensionHash, getLogs } from '@colony/colony-js';
 
+import { addVotingReputationParamsToDB } from '~handlers/extensions';
```
Comment (jakubcolony) [inline src/trackExtensions.ts:3]: Nitpick, but having to import it from `~handlers` suggests to me it could live in `~utils` instead

**Review by jakubcolony — APPROVED** [review]
All good, one thing that relates more to the CDapp part of this (that I was too slow to review) is that we could use a union type for extension params instead, just to reduce the nesting (and since there can only be one type of params for an extension), e.g.:

```gql
type VotingReputationParams {
  ...
}

union ExtensionParams = VotingReputationParams | OtherParamsInTheFuture;

type ColonyExtension {
  ...
  params: ExtensionParams
}
```

**Comment by jakubcolony** [comment]
@willm30 Was that not the case only for models?

**Comment by jakubcolony** [comment]
I got the schema to compile with a union. Haven't tried actually querying it though

**Comment by willm30** [comment]
@jakubcolony Not certain but even on non-models, we'd still need to write some logic to tell Amplify which type it resolves to given the surrounding data (in this case, the extension hash).

**Pull request merged**